packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -153,8 +161,54 @@
       };
     }
 
-    protected mounted(): void {
-      this.$x.emit('QueryPreviewRequestChanged', this.queryPreviewRequest);
+    /**
+     * The debounce method to trigger the request after the debounceTimeMs defined.
+     *
+     * @returns The search request object.
+     * @internal
+     */
+    protected get emitQueryPreviewRequestChanged(): DebouncedFunction<[SearchRequest]> {
+      return debounce(request => {
+        this.$x.emit('QueryPreviewRequestChanged', request);
+      }, this.debounceTimeMs);
+    }
+
+    /**
+     * Initialises watcher to emit debounced requests, and first value for the requests.
+     *
+     * @internal
+     */
+    protected created(): void {
+      this.$watch(
+        () => this.queryPreviewRequest,
```

Comment:
Any reason to use the function form instead of `'queryPreviewRequest'`?

Reply:
Not losing type safety 🤣:

![image](https://user-images.githubusercontent.com/68222542/193257510-600d0eaa-40cb-4bea-a33d-a4332025475c.png)